examples/js/controls/TransformControls.js

```diff
@@ -128,6 +128,21 @@ THREE.TransformControls = function ( camera, domElement ) {
 		domElement.removeEventListener( "touchcancel", onPointerUp );
 		domElement.removeEventListener( "touchleave", onPointerUp );
 
+		this.traverse( function( object ) {
+
+			if ( object.material ) {
+
+				object.material.dispose();
```

Comment:
This is going to crash on array materials.

Reply:
Disposing objects needs to be consolidated into a single place. For example, this is what I currently have and I re-use it everywhere: https://github.com/infamous/infamous/blob/39ab9e3854f32911c1bc53ac0dca0d1696f4242d/src/utils/three.ts#L10-L61

It is better than writing duplicate dispose code all over the place.

And also, now that I realize based on discussion in https://github.com/mrdoob/three.js/issues/15743, this does not dispose textures (which may be something you intend to do in some cases).

Reply:
> This is going to crash on array materials.

`TransformControls` does not assign multiple materials to its renderable 3d (helper) objects so this should be no issue.

Reply:
> Disposing objects needs to be consolidated into a single place.

I think it's better to define just an interface for disposal operations ( `dispose()`) and let the respective class decide how to free things.